lib/Controller/PersonalSettingsController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types=1);
+
+namespace OCA\GPodderSync\Controller;
+
+use GuzzleHttp\Psr7\BufferStream;
+use GuzzleHttp\Psr7\StreamWrapper;
+use OCA\GPodderSync\Core\PodcastData\PodcastDataReader;
+use OCA\GPodderSync\Core\PodcastData\PodcastMetricsReader;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\StreamResponse;
+use OCP\AppFramework\OCS\OCSException;
+use OCP\Http\Client\IClient;
+use OCP\Http\Client\IClientService;
+use OCP\IRequest;
+
+class PersonalSettingsController extends Controller {
+
+	private string $userId;
+	private PodcastMetricsReader $metricsReader;
+	private PodcastDataReader $dataReader;
+
+	// TODO: Use httpClient via PodcastDataReader instead
+	private IClient $httpClient;
+
+	public function __construct(
+		string $AppName,
+		IRequest $request,
+		string $UserId,
```

Comment:
I noticed this as well in some testing but never changed it. Nice catch. I changed to `?string` instead, to allow nullable strings.

Reply:
`?string` in constructor parameters sounds good, but everywhere else it's not necessary, since it's catched by `$this->userId = $UserId ?? '';`.  
And the handling of unauthorized calls is unnecessary, Nextcloud already does this on its own:
```
curl http://localhost:8080/index.php/apps/gpoddersync/personal_settings/metrics
{"message":"Current user is not logged in"}
```

Reply:
Ah sweet, didn't know that. Seemed weird that it hit the controller's constructor when unauthorized but wouldn't hit the endpoint methods, but thx for clearing that up :+1: